docs/source/contributor-guide/index.md

```diff
@@ -72,13 +72,35 @@ recently active community members in the ticket to get their attention.
 
 ### What Features are Good Fits for DataFusion?
 
-DataFusion is designed to highly extensible, and many features can be implemented
-as extensions without changing the core of DataFusion.
-
-We are [working on criteria for what features are good fits for DataFusion], and
-will update this section when we have more to share.
-
-[working on criteria for what features are good fits for datafusion]: https://github.com/apache/datafusion/issues/12357
+DataFusion is designed to highly extensible, and many features can be
+implemented as extensions without changing the core of DataFusion. Support for
+new functions, data formats, and similar functionality can be added using those
+extension APIs, and there are already many existing community supported
+extensions listed in the [extensions list].
+
+Query engines are complex pieces of software to develop and maintain. Given our
+limited maintenance bandwidth, we try to keep the DataFusion core as simple and
+focused as possible, while still satisfying the [design goal] of an easy to
+start initial experience.
+
+With that in mind, contributions that meet the following criteria are more likely
+to be accepted:
+
+1. Bug fixes for existing features
+2. Test coverage for existing features
+3. Documentation improvements / examples
+4. Performance improvements to existing features (with benchmarks)
+5. "Small" functional improvements to existing features
+6. Additional APIs for extending DataFusion's capabilities
+
+Contributions that likely require discussion prior to acceptance include:
```

Comment:
We probably should note the preferred way to discuss, GH Discussion/ASF slack, etc